### DISCUSSION TITLE
Erro MAPOS ao instalar em servidor hostgator com PHP 8

### DISCUSSION THREAD

**Discussion by wanderleyrodrigo** [RamonSilva20/mapos#2029] [Q&A]
Boa noite, inicialmente parabéns pelo projeto.
Estou utilizando em uma estação local e agora quero migrar para um servidor web na Hostgador, fiz todos os procedimentos e quando vou acessar o mesmo apresenta o erro abaixo.

 Whoops \ Exception \ ErrorException (E_WARNING)
Undefined variable $configuration
![Captura de tela de 2023-01-08 19-08-43](https://user-images.githubusercontent.com/122235964/211221514-94b3f927-4243-425d-838a-6657f0d8d04f.png)

**Comment by andremp15** [comment]
Oi boa noite, você está utilizando o hostgator já com outro site se a resposta for sim e ele estiver funcionando normalmente o problema tá na instalação se a resposta que você não tem outro site e você contratou recentemente a hospedagem liga para assistência técnica que ele te dá o suporte necessário é um erro do próprio hostgator

**Reply by wanderleyrodrigo** [reply]
Opa bom dia, sim tenho um site já funcional e rodando.
Fiz a instalação nova do MapOS e mesmo assim permaneceu o erro, estranho.
Vou tentar ver com o suporte se é algo do servidor PHP, mas tenho 3 sites rodando normalmente lá.
Valeu pelo retorno.

**Comment by Henrique-Miranda** [comment]
Você rodou o comando `composer install --no-dev` na pasta oficina? Se sim rodou sem erros?

**Reply by wanderleyrodrigo** [reply]
Sim Henrique, rodei o comando e rodou perfeito sem erros.

**Comment by Jhenriquek1** [comment]
Obrigado pela atenção descobri que  a versão do xamp estava errada resolvido

Enviado do Email<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows

De: Professor Wanderley ***@***.***>
Enviado:sexta-feira, 13 de janeiro de 2023 06:45
Para: ***@***.***>
***@***.***>
Assunto: Re: [RamonSilva20/mapos] Erro MAPOS ao instalar em servidor hostgator com PHP 8 (Discussion #2029)


Sim Henrique, rodei o comando e rodou perfeito sem erros.

**Comment by allsilva** [comment]
Recentemente instalei o linux mint e ai a incompatibilidade do PHP8...

➜  composer install --no-dev
Installing dependencies from lock file
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - fzaninotto/faker is locked to version v1.9.2 and an update of this package was not requested.
    - fzaninotto/faker v1.9.2 requires php ^5.3.3 || ^7.0 -> your php version (8.1.2) does not satisfy that requirement.
  Problem 2
    - piggly/php-pix is locked to version 1.2.8 and an update of this package was not requested.
    - piggly/php-pix 1.2.8 requires php ^7.2 -> your php version (8.1.2) does not satisfy that requirement.

**Reply by willph** [reply]
Versão do PHP ideal é 7.4.xx.

Php na versão 8 ainda não funciona.

**Reply by locopine** [reply]
> Recentemente instalei o linux mint e ai a incompatibilidade do PHP8...
> 
> ➜ composer install --no-dev Installing dependencies from lock file Verifying lock file contents can be installed on current platform. Your lock file does not contain a compatible set of packages. Please run composer update.
> 
> Problem 1 - fzaninotto/faker is locked to version v1.9.2 and an update of this package was not requested. - fzaninotto/faker v1.9.2 requires php ^5.3.3 || ^7.0 -> your php version (8.1.2) does not satisfy that requirement. Problem 2 - piggly/php-pix is locked to version 1.2.8 and an update of this package was not requested. - piggly/php-pix 1.2.8 requires php ^7.2 -> your php version (8.1.2) does not satisfy that requirement.

------------------------------------------------------------------------------------
Bom dia, @allsilva  e @willph!

Sou utilizador do Linux Mint e foi possível rodar a aplicação efetuando as seguintes alterações:

- apaguei o arquivo "composer.lock";
- alterei as versões dos arquivos apontados, no **_composer.json_**, da seguinte forma

  1. "piggly/php-pix": "^1.2.5" para "piggly/php-pix": "^2.0.2"; e
  2. "fzaninotto/faker": "^1.9" para "fzaninotto/faker": "*"

- executei o comando "composer update --no-dev";
- e, para sair do "exception" de "Undefined variable $configuration" alterei o  metodo index() em "application/controllers/Login.php para o seguinte:

```
public function index()
   {
       $this->load_configuration();
       $this->load->view('mapos/login', ['configuration' => $this->data['configuration']]);
   }
```

e inclui o método "load_configuration()" no final da classe Login:

```
    private function load_configuration()
    {
        $this->CI = &get_instance();
        $this->CI->load->database();
        $configuracoes = $this->CI->db->get('configuracoes')->result();

        foreach ($configuracoes as $c) {
            $this->data['configuration'][$c->config] = $c->valor;
        }
    }
```

Em suma, até que o autor compatibilize a aplicação para a versão PHP 7.0 || 8.0 || > é uma saída para quem quiser usar em versão PHP maior que 7.

Sucesso à todos.
